src/scenes/Root/CommentsBar/CommentItem/CommentItem.tsx

```diff
@@ -0,0 +1,179 @@
+import { useMutation } from '@apollo/client';
+import { MoreVert } from '@mui/icons-material';
+import { Box, IconButton, MenuProps, Paper, Typography } from '@mui/material';
+import Blocks from 'editorjs-blocks-react-renderer';
+import { useState } from 'react';
+import { removeItemFromList } from '~/api';
+import { useDateTimeFormatter } from '~/components/Formatters';
+import { CommentItemMenu } from '../CommentItemMenu';
+import { CommentReply } from '../CommentReply';
+import {
+  CommentPropsFragment,
+  CommentThreadPropsFragment,
+  DeleteCommentDocument,
+} from '../CommentsBar.graphql';
+import { useCommentsContext } from '../CommentsBarContext';
+
+export interface CommentProps {
+  comment: CommentPropsFragment;
+  repliesCount?: number;
+  isChild?: boolean;
+  resourceId: string;
+  isExpanded?: boolean;
+  parent: CommentThreadPropsFragment;
+  handleDeleteComment?: (comment: CommentPropsFragment) => Promise<void> | void;
+  handleEditComment?: (comment: CommentPropsFragment) => Promise<void> | void;
```

Comment:
callback properties always start with `on` not `handle`. Handle is the prefixed used to declare the handler function for the callback property, though I don't really care if this is followed. I think I define mine inline a lot of times anyways.

Since this is a comment as well I think the comment suffix is superfluous. So `onDelete`/`onEdit` would be better. Though it looks like edit isn't even used?

Also the `Promise` return type isn't the best. This code that calls the function doesn't await the promise before doing anything; it just calls it and leaves the promise hanging. So it would be better to "hang" that promise in the calling code. It's more explicit that it is async logic that is not awaited.